autrainer/datasets/msppodcast.py

```diff
@@ -0,0 +1,202 @@
+            stratify: Columns to stratify the dataset on. Defaults to None.
+            standardize: Whether to standardize the data. Defaults to False.
+            categories: used to filter out specific emotional categories. 
+                Useful for training on subset of data/classes, such as the classic
+                ["A", "H", "N", "S"] 4-class problem found in literature. 
+                Defaults to None.
+        """
+        # TODO: expand for multitask regression
+        assert target_column in ("EmoClass", "EmoAct", "EmoVal", "EmoDom"), (
+            f"{target_column} not included in target list, please choose one of: "
+            "[EmoClass, EmoAct, EmoVal, EmoDom]"
+        )
+        task = "classification" if target_column == "EmoClass" else "regression"
+        self.categories = categories
+        super().__init__(
+            task=task,
+            path=path,
+            features_subdir=features_subdir,
+            seed=seed,
+            metrics=metrics,
+            tracking_metric=tracking_metric,
+            index_column=index_column,
+            target_column=target_column,
+            file_type=file_type,
+            file_handler=file_handler,
+            batch_size=batch_size,
+            inference_batch_size=inference_batch_size,
+            train_transform=train_transform,
+            dev_transform=dev_transform,
+            test_transform=test_transform,
+            stratify=stratify,
+        )
+        self.standardize = standardize
+        if self.standardize:
+            train_data = torch.cat([x for x, *_ in self.train_dataset])
+            standardizer = Standardizer(
+                mean=train_data.mean(0).tolist(),
+                std=train_data.std(0).tolist(),
+            )
+
+            self.train_transform += standardizer
+            self.dev_transform += standardizer
+            self.test_transform += standardizer
+
+            self.train_dataset = self._init_dataset(
+                self.df_train, self.train_transform
+            )
+
+    @staticmethod
+    def download(path: str) -> None:  # pragma: no cover
+        """
+        Download the MSP-Podcast dataset.
+
+        As this dataset is not publicly-available, please download it manually
+        by contacting Prof. Carlos Busso: 
+        https://ecs.utdallas.edu/research/researchlabs/msp-lab/MSP-Podcast.html
+
+        This function will not do anything.
+
+        For more information on the data, see:
+        https://doi.org/10.1109/TAFFC.2017.2736999
+        """
+        return None
+    
+    def load_dataframes(
+        self,
+    ) -> Tuple[pd.DataFrame, pd.DataFrame, pd.DataFrame]:
+        """Load the dataframes.
+
+        Overrides base class so there is no need
+        to create separate train/dev/test files.
+
+        Returns:
+            Dataframes for training, development, and testing.
+        """
+        df = pd.read_csv(os.path.join(self.path, "Labels", "labels_consensus.csv"))
```

Comment:
All CSVs of the dataset have to be located directly in the `${path}` directory. But its totally fine to construct the subsets from `labels_consensus.csv`.

Reply:
You mean I need to save them for the preprocessing to work, right?